docs/design-documents/tools/cmake.md

```diff
@@ -1,101 +1,118 @@
+The build profiles such as release or debug are defined in the scripts found in `cmake/profiles/`.
 
+### 4. MCU Core CMake Scripts
 
-`add_subdirectory` always adds the directory to the main CMake.
+The MCU core definitions are defined in the scripts found in `cmake/cores/`.
 
-```
-add_subdirectory(drivers)
-```
+### 5. Utilities CMake Scripts
 
-Conditionally directories addition based on the config:
+Custom functions/macros used within Mbed OS.
 
-```
-add_subdirectory_if_config(CONFIG_REQUIRES_RTOS rtos)
-```
-
-Conditionally directories addition based on the target/toolchain/feature.
+### 6. Application CMake
 
+The CMake script that must be included by all applications using:
 
 ```
-add_subdirectory_if_target(STM targets\TARGET_STM)
-add_subdirectory_if_toolchain(GCC_ARM cmsis\TARGET_GCC_ARM)
-add_subdirectory_if_feature(BLE ble\FEATURE_BLE)
+include(${MBED_ROOT}/cmake/app.cmake)
 ```
 
+### 7. Component `CMakeLists.txt` Entry Point
 
-## Application CMake
+This file statically defines the build specification of an Mbed OS component. It contains conditional statements that depend on the configuration parameters generated by `mbed-tools`. The component entry points make use of functions/macros defined in `util.cmake` to conditionally include or exclude directories.
+The rule of thumb is to not expose header files that are internal. We would like to avoid having everything in the include paths as we do now.
 
-We should provide application CMake functionality with  our own configuration. There are couple of approaches we could take. Statically defined CMake but then this disconnectes config and CMake - as CMake contains configuration for a project (like includes, sources, etc). Our build tool would need to parse CMake to get all paths used in the project or Mbed OS to find out where to look for configuration file. Therefore the build system has a knowledge as it is currently. We use `requires` to include/exclude modules.
+## Building an Application
 
-By default, baremetal would be selected - requires set to hal, platform, drivers and cmsis. If an app needs anything else, would use `requires` in the config to include - BLE/networking/etc.
+`mbed-tools` is the next generation of command line tooling for Mbed OS. `mbed-tools` replaces `mbed-cli` and the Python modules in the `mbed-os/tools` directory.
+
+`mbed-tools` consolidates all of the required modules to build Mbed OS, along with the command line interface, into a single Python package which can be installed using standard Python packaging tools.
 
 A user create own CMake file to configure an application, also with `mbed_app.json` configuration file. The building of an app would look like:
```

Comment:
```suggestion
Each application contains a top-level CMakeLists.txt file. The `mbed-tools init` command can create this top-level CMakeLists.txt, or a user can create it manually. Each application also has a number of json configuration files. `mbed-tools configure` creates an Mbed configuration CMake file (`.mbedbuild/mbed_config.cmake`). The process for building an application looks like:
```

Reply:
I would go with "Each application contains a top-level `CMakeLists.txt` entry point" - this is how the CMake documentation describes this source file type https://cmake.org/cmake/help/latest/manual/cmake-language.7.html#directories. It's also consistent with how we refer to them earlier in the design doc.